plugins/chat/spec/services/chat/update_user_channel_membership_spec.rb

```diff
@@ -0,0 +1,88 @@
+# frozen_string_literal: true
+
+RSpec.describe Chat::UpdateUserChannelMembership do
+  describe described_class::Contract, type: :model do
+    it { is_expected.to validate_presence_of(:channel_id) }
+  end
+
+  describe ".call" do
+    subject(:result) { described_class.call(params:, **dependencies) }
+
+    fab!(:current_user, :user)
+    fab!(:channel, :category_channel)
+    fab!(:membership) do
+      Fabricate(
+        :user_chat_channel_membership,
+        user: current_user,
+        chat_channel: channel,
+        pinned: false,
+      )
+    end
+
+    let(:params) { { channel_id: channel.id, pinned: true } }
+    let(:dependencies) { { guardian: current_user.guardian } }
+
+    context "when contract is invalid" do
```

Comment:
Usually here, the idea is to simply check the service has a contract defined and that it will stop its execution if an attribute is invalid. Then you can do all the validations and edge-case scenarios in the contract examples (on lines 4-6).